### PR TITLE
CSS clean up 🎉

### PR DESCRIPTION
**Pull request by sethmcleod**
##### Description
I was working on #430 and spent some time cleaning up font rules and general CSS. After a few hours I realized I had already touched more than a few files that were unrelated to the modals. So I decided to break off the CSS polish into a separate PR.

Mostly I made sure all of our CSS files adhered to the style lint rules, but here are some specifics:

- Fixed duplicate rules and properties
- Fixed general typos like extra semicolons
- Make sure font families have generic in stack
- Added units to whitelist: vw, vh, deg, pt, vmin
- Prefer single quotes over double
- Adhere to 4 space, tab indentation
- Add newline between rules
- End files with newline

I know there's a lot of code to review, but the majority of these changes are just indentation updates and rule organization to adhere to `no-descending-specificity`. I reviewed most pages of the app and didn't see anything out of place due to these updates.

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

### CODE REVIEW

**Comment by thelostone-mc** [comment]
@sethmcleod I remember a while back there was a decision to use 2 spaces as against to 4 !
Did that change ? 🤔

**Comment by sethmcleod** [comment]
Thanks for the review @thelostone-mc!

- I chose to use 4 space, tab indentation because that's what was detailed in the `.editorconfig` and `.stylelintrc` files, _but_ I believe 2 spaces (no tab character) works best for CSS and I'd be happy to make that update.
- Ending files with a newline is a practice that's used throughout the codebase and also detailed in the `.editorconfig` file.
- I did double-check those font declarations.
- I did not modify the bootstrap CSS but I did add a comment to disable stylelint in the minified file. I'll revert that change so this is easier to review.

**Comment by sethmcleod** [comment]
Just pushed an update to go with 2 space indentation for CSS files. Also adjusted the `.editorconfig` and `.stylelintrc` files to reflect this.

File: app/assets/v2/css/jquery-ui.css [hunk 1]
```diff
@@ -966,346 +1106,920 @@ a:visited.ui-button,
...
+
 .ui-state-error .ui-icon,
 .ui-state-error-text .ui-icon {
-    background-image: url("images/ui-icons_cc0000_256x240.png");
+  background-image: url("images/ui-icons_cc0000_256x240.png");
 }
+
 .ui-button .ui-icon {
-    background-image: url("images/ui-icons_777777_256x240.png");
+  background-image: url("images/ui-icons_777777_256x240.png");
```
[the remark this answers is not in the document]
Comment (sethmcleod) [reply app/assets/v2/css/jquery-ui.css]: Ooops 🙄will fix that up

File: app/assets/v2/css/ios.css [hunk 2]
```diff
@@ -176,11 +178,9 @@
     text-align: center;
     margin: 15px 40% 0;
   }
-
   .hidden-lg {
     display: none;
   }
-
```
[the remark this answers is not in the document]
Comment (sethmcleod) [reply app/assets/v2/css/ios.css]: Yes, I'll make sure all rules nested in media blocks have empty lines between them.

File: app/assets/v2/css/external_bounties/bounty.css [hunk 3]
```diff
@@ -91,32 +91,35 @@
...
 
-pre{
-    white-space: pre-wrap;       /* Since CSS 2.1 */
-    white-space: -moz-pre-wrap;  /* Mozilla, since 1999 */
-    white-space: -pre-wrap;      /* Opera 4-6 */
-    white-space: -o-pre-wrap;    /* Opera 7 */
-    word-wrap: break-word;     
+pre {
+  white-space: pre-wrap;
+  /* Since CSS 2.1 */
```
[the remark this answers is not in the document]
Comment (sethmcleod) [reply app/assets/v2/css/external_bounties/bounty.css]: I'm cool with removing these types of comments, prefixed stuff is pretty self explanatory.

**Comment by sethmcleod** [comment]
I've done some refactoring based on your feedback; mostly just adding newlines to rules in _@media_ and _@keyframes_ blocks.

Also I truly apologize for how many lines there are to review 😓Thanks for taking all this on!

**Comment by thelostone-mc** [comment]
Will get this checked out this week so as to reduce the chances of conflicts ^_^

File: .editorconfig [hunk 4]
```diff
@@ -9,7 +9,6 @@ indent_style = space
 insert_final_newline = true
 trim_trailing_whitespace = true
 
-
 # 4 space indentation
-[*.{css,scss,py}]
+[*.{py}]
```
Comment (mbeacom) [inline .editorconfig]: 👍

File: .stylelintrc [hunk 5]
```diff
@@ -9,14 +9,21 @@
         ]
       }
     ],
-    "indentation": "tab",
+    "indentation": 2,
```
Comment (mbeacom) [inline .stylelintrc]: 👍

**Comment by thelostone-mc** [comment]
And there comes the conflicts 😓

**Comment by sethmcleod** [comment]
I can handle these conflicts if you'd like.

**Comment by thelostone-mc** [comment]
@sethmcleod  Wanna wait until @mbeacom is ready to merge ? 
Else we might have to resolve conflicts a second time :P

**Comment by sethmcleod** [comment]
Sure, no worries! Just let me know when and I'll get this PR cleaned up.

**Comment by mbeacom** [comment]
@sethmcleod Good to go! Feel free to resolve the conflicts and I'll get this merged.

Thanks again!

**Comment by sethmcleod** [comment]
Hmm, odd. I rebased against remote _master_ then merged it into my branch, but I'm not seeing these conflicts. I can resolve them through this site, but I'm curious if there was something I did wrong here:

```bash
git checkout master
git rebase gitcoinco/master
git checkout css-cleanup
git merge master
```

Any ideas @mbeacom?

**Comment by thelostone-mc** [comment]
```
git fetch gitcoinco master
git rebase gitcoincoaster
```

^ That should throw I'm the conflicts

**Comment by owocki** [comment]
> Any ideas @mbeacom?

you probably need to push your master upstream to github so that the conflicts are updated here

**Comment by sethmcleod** [comment]
There we go, thanks for that 👍

**Comment by mbeacom** [comment]
I usually go the merge route and maintain an updated `master` branch in my fork:

```shell
git checkout master
git fetch upstream
git merge upstream/master
git push origin master
git checkout my-branch
git merge master
git push
```

Similar to the above rebase:

```shell
git checkout my_branch
git fetch upstream
git merge upstream/master
```

https://help.github.com/articles/syncing-a-fork/

**Pull request merged**